blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+
+```robot
+SBO004.001 Attempt to boot file without the key from Shell (firmware)
+    [Documentation]    This test verifies that Secure Boot blocks booting a file
+    ...    without a key.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO004.001 not supported
+    Mount ISO As USB    ${CURDIR}/../scripts/secure-boot/images/NOT_SIGNED.img
+    # 1. Make sure that SB is enabled
+    Power On
+    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
+    Enable Secure Boot    ${sb_menu}
+    # Save Changes And Reset
+    # Changes to Secure Boot menu takes action immediately, so we can just reset
+    Reset System
+    Boot Efi File Should Fail    hello.efi    NOT_SIGNED
+```
+
+As you can see, the test case is brief and easily readable thanks to its use of
+predefined keywords.
+
+The test suite can be found in the
+[OSFV repository](https://github.com/Dasharo/open-source-firmware-validation/blob/main/dasharo-security/secure-boot.robot)
+and the detailed description of each test is available on the
+[Unified Test Documentation](https://docs.dasharo.com/unified-test-documentation/dasharo-security/206-secure-boot)
+page.
+
+The test suite can be run using the following command:
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:<rte_ip> \
+  -v snipeit:no \
+  -v config:mpl-pip4 \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
+
+To use QEMU execute this instead:
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:127.0.0.1 \
+  -v snipeit:no \
+  -v config:qemu \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
+
+### Results
+
+The tests were performed on MPL PIP44. The following images show the test suite
+results:
+![uefi-sb-results-report](/img/uefi-sb-results-pt1.png)
+![uefi-sb-results-report](/img/uefi-sb-results-pt2.png)
+
+The results lead to the following conclusions:
+
+- The state of UEFI Secure Boot functionality can be freely modified from the
+UEFI BIOS Menu, and it is correctly detected from the operating system.
+- Verification of launched images works correctly when UEFI Secure Boot is
+enabled. The firmware allows the execution of files signed with the appropriate
+keys but blocks the booting of unsigned files or files signed with keys not
+present in the database (DB) or files with hashes not stored in the database
+(DB).
+- The firmware correctly recognizes the chain of trust when intermediate
+certificates are used, allowing their use in the verification process.
+- Keys intended for UEFI Secure Boot must be generated using the RSA
+cryptographic algorithm and sizes 2048, 3072, and 4096. Keys generated with the
+ECDSA cryptographic algorithm are not correctly supported.
+- The firmware does not verify the expiration date of certificates during the
+verification of launched files. It means that enrolled certificates may expire
+and will not affect the ability to boot files verified by them.
```

Comment:
What are the potential consequences of that?

Reply:
Done ad87522d2cb6e682dc5e76ebb3c26599f8ec764c